src/utils/styling-utils.js

```diff
@@ -28,3 +30,10 @@
 export function getCombinedClassName(...classNames) {
   return classNames.filter(cn => cn && typeof cn === 'string').join(' ')
 }
+
+export function styledWithClass(elem, className, styleObj) {
+  // eslint-disable-next-line react/display-name
+  return withWrapper(styled(elem, styleObj), Styled => props => (
+    <Styled className={className} {...props} />
```

Comment:
It would be interesting to take a deep dive on the performance between hooks and nested components.

This is the component tree using the method that you have above
![image](https://user-images.githubusercontent.com/1887318/82243452-a3cd8080-990d-11ea-98cb-e117619a048f.png)

Changing it to use `useStyletron` with this
```
export function styledWithClass(elem, definitionClassName, styleObj) {

  const Component = elem;
  const Styled = props => {
    const [css] = useStyletron();
    const {className, style, ...rest} = props;
    const styleClass = css({...styleObj, ...style});
    return <Component className={getCombinedClassName(styleClass, definitionClassName, className)} {...rest} />
  }
  Styled.displayName = `styled(${getDisplayName(elem)})`
  return Styled;
}
```
we can get the component tree to look like
![image](https://user-images.githubusercontent.com/1887318/82243569-daa39680-990d-11ea-9fe3-be1ad5057cfe.png)

Reply:
as a point of reference, these are the hooks that are being used
![image](https://user-images.githubusercontent.com/1887318/82245343-c8772780-9910-11ea-8541-a19accb9840d.png)